### ISSUE TITLE
Error with the Vulkan-renderers Forward+ and Mobile: game window shows black and closes without errors shortly after

### ISSUE THREAD

**Issue by CodeStrix** [godotengine/godot#91373]
### Tested versions

Reproducable in 4.2.2 stable (c# variant)

### System information

Windows 10 (fully updated), Godot 4.2.2 stable, Forward+/Mobile

### Issue description

When I try to run a scene with only the 2d root node, the game window shows up, stays black and closes without any visual sign of the game (testet in compatibility mode - works fine, so there is actually an problem). I run an old (but to now flawlessly working) GTX 1070 with fully updated drivers (as my windows is). The Vulkan api is working fine with other games, that rely on it - I therefore assume, its something Godot-related.
I made several new projects with different root nodes - nothing works. Neither with some assets loaded nor without them. 

### Steps to reproduce

No need for stept - always happens. 

### Minimal reproduction project (MRP)

C @ Steps to reproduce.

**Comment by clayjohn** [comment]
Can you try running Godot from the command line so that you can see if any errors are printed out?

The command to use is:
```
./Godot_v4.2.2-stable_win64.exe --verbose
```

**Comment by clayjohn** [comment]
In this case it seems to be the same issue as https://github.com/godotengine/godot/issues/88991 which was fixed by removing the twitch overlay the person had installed.

From the error it seems like the Vulkan loader is looking for the epic games overlay but not finding it. Did you uninstall or reinstall the Epic Games Launcher recently?

**Comment by CodeStrix** [comment]
> In this case it seems to be the same issue as #88991 which was fixed by removing the twitch overlay the person had installed.
> 
> From the error it seems like the Vulkan loader is looking for the epic games overlay but not finding it. Did you uninstall or reinstall the Epic Games Launcher recently?
> 
> In either case, doing a clean uninstall reinstall of the launcher might fix the issue

The last time I touched the Epic Launcher Installment was years ago. But since there have been Updates of the Launcher - who might know. I'll try the clean un-/reinstall and shall report ;).

**Comment by CodeStrix** [comment]
Clean de-/reinstall of EPIC Launcher - no luck. The same problem and error message persists. But i noticed, that the error message is pointing to a wron directory for the epic launcher. So I reinstalled the Launcher in the folder from the error message - still no progress and the same problem and error message persists.

**Comment by akien-mga** [comment]
You might need to go through the Windows registry to locate where it installed a layer that's referencing obsolete files from Epic Online Services and GOG Galaxy, judging by the names.

See https://github.com/godotengine/godot/issues/56089#issuecomment-1172922023 for potential locations for these registry keys.

It's a flaw of the Vulkan ecosystem that any application can register implicit or explicit layers and typically doesn't properly clean after themselves, impacting all other Vulkan apps.

**Comment by CodeStrix** [comment]
So the good news is, that getting rid of the anyway obsolete registry keys actually resolved the errors. 

Bad news is - nothing changed. I now don't have any errors being thrown - but still cannot use forward+ or mobile to actually run any scene/game (I have yet to restart....).

**Comment by CodeStrix** [comment]
So i put in the whole log when starting the sample scene (2D node, packed_scene with animated script, collider and camera2D - no own scripts so far). Maybe someone can find a hint in the log.

**Comment by bitmammoth** [comment]
I was able to resolve this for myself by uninstalling GOG Galaxy app and then removing that registry entry.
hint was this line and the name in the registry:
Layer name GalaxyOverlayVkLayer_VERBOSE does not conform to naming standard (Policy #LLP_LAYER_3)

Located here:
Computer\HKEY_LOCAL_MACHINE\SOFTWARE\Khronos\Vulkan\ImplicitLayers

**Comment by CodeStrix** [comment]
I have updated the Godot version from 4.2.2 to 4.3 and although there is still the same problem, verbose shows some new lines in the log:
`================================================================
CrashHandlerException: Program crashed with signal 11
Engine version: Godot Engine v4.3.stable.mono.official (77dcf97d82cbfe4e4615475fa52ca03da645dbd8)
Dumping the backtrace. Please include this when reporting the bug to the project developer.
[1] error(-1): no debug info in PE/COFF executable
[2] error(-1): no debug info in PE/COFF executable
[3] error(-1): no debug info in PE/COFF executable
[4] error(-1): no debug info in PE/COFF executable
[5] error(-1): no debug info in PE/COFF executable
[6] error(-1): no debug info in PE/COFF executable
[7] error(-1): no debug info in PE/COFF executable
[8] error(-1): no debug info in PE/COFF executable
[9] error(-1): no debug info in PE/COFF executable
[10] error(-1): no debug info in PE/COFF executable
[11] error(-1): no debug info in PE/COFF executable
[12] error(-1): no debug info in PE/COFF executable
[13] error(-1): no debug info in PE/COFF executable
[14] error(-1): no debug info in PE/COFF executable
[15] error(-1): no debug info in PE/COFF executable
[16] error(-1): no debug info in PE/COFF executable
[17] error(-1): no debug info in PE/COFF executable
[18] error(-1): no debug info in PE/COFF executable
[19] error(-1): no debug info in PE/COFF executable
[20] error(-1): no debug info in PE/COFF executable
[21] error(-1): no debug info in PE/COFF executable
-- END OF BACKTRACE --
================================================================`

I have updated the thread start accordingly.

**Comment by Zireael07** [comment]
This backtrace is useless due to no debug info.

**Comment by CodeStrix** [comment]
I would argue that the lack of debug info also might tell something :).

**Comment by Zireael07** [comment]
@CodeStrix No, because no debug info is a feature of the Godot build you're using, it's not helpful at all

**Comment by CodeStrix** [comment]
So I am able to turn on the debug info?

**Comment by Zireael07** [comment]
Yes, or try those https://github.com/Calinou/godot-debug-builds/releases

**Comment by CodeStrix** [comment]
**So I finally have been able to resolve the problem.** When installing Vulkan via NVIDIA Geforce Experience, a neat little program called ReShade is installed as well. One of the ReShade libraries actually lead to the Godot project failure. By just deleting the corresponding ReShade folder (seems to be installed without reg entries) I have been able to resolve my issue. 

I will keep the thread open - in case someone actually knowledgeable wants to add something important to consider.

**Comment by clayjohn** [comment]
Thank you for updating the post with your findings!

We should see if there is anything we can do to force-disable reshade to avoid this problem for others in the future

**Comment by CodeStrix** [comment]
@clayjohn you and all potential beneficiaries are welcome. Glad to have this issue "kinda" resolved.
